### DISCUSSION TITLE
How might we coordinate (starting to) deliver cool features to users?

### DISCUSSION THREAD

**Discussion by bollwyvl** [jupyterlite/jupyterlite#103] [General]
So we're building up a backlog of pretty interesting stuff, getting the demo pretty good, and starting to make some impact on some upstreams, all of which is very cool.

It seems like getting to something _shippable_, at least as a pre-release, is very close at hand, but assessing what to work on _next_ is getting a little peril-fraught. I guess some I'm hoping for some discussion here as to _how_ we get to making that easier for existing and future contributors, not the actual set of features, which presumably that mechanism would support. 

I haven't worked with this feature on github much, but I'll try making some replies, and maybe this will make more sense....

# Propose new organization ideas below!

and/or

# :arrow_up: organization ideas below

**Comment by bollwyvl** [comment]
## Use Projects

GitHub projects are pretty good, and offer some nice features. Start one project, have some kanban alleys, and away we go.

**Comment by bollwyvl** [comment]
## Use Milestones

These have been around for  a long time, and are generally good enough to plan for NEXT, NEXT+1 kind of features, and the little progress bar feels pretty good. They don't really have prioritization, that I can tell, however, so kinda don't help deciding on the next thing.

**Comment by bollwyvl** [comment]
## Use a ROADMAP

Having a ROADMAP.md can be effective, as tables, checklists, etc. are usually "good enough" for light project planning. Because they are backed by PRs, there's plenty of places to discuss, and because the roadmap gets check into git, the provenance is of course excellent.

**Comment by bollwyvl** [comment]
## Use Discussions

As the newest feature, discussions seem kinda weird, but are threaded (which helps longer discussions), can be pinned, have voting. It looks like an _individual comment thread_ can't be promoted to an issue, though.

**Comment by jtpio** [comment]
Thanks Nick for starting this discussion.

Maybe milestones and issues are enough for now. We could also try a project if we use it for something else than the classic "todo -> in progress -> done" workflow.

For now I guess the plan is to have a first MVP (0.1.0) with the following:

- publish packages to `npm`, PyPI (and GH releases)
- tooling to create lab and serverlite extensions: https://github.com/jtpio/jupyterlite/issues/41, https://github.com/jtpio/jupyterlite/issues/55
- good deployment docs and tooling: https://github.com/jtpio/jupyterlite/issues/67, https://github.com/jtpio/jupyterlite/issues/66
- update to the 3.1 lab alpha packages to enable RTC (#89)

Maybe we should list these items in an issue, or create an MVP project?

In parallel, folks could also contribute to nice-to-have features or bug fixes, since they should normally be self-contained:

- issues with the `bug` label
- things like https://github.com/jtpio/jupyterlite/issues/31, https://github.com/jtpio/jupyterlite/issues/18, https://github.com/jtpio/jupyterlite/issues/9, https://github.com/jtpio/jupyterlite/issues/60

**Reply by bollwyvl** [reply]
Welp, if nobody's _against_ project, I think they provide about the best data/workflow, even though that data doesn't live in git itself.
